### PR TITLE
Automatically resolve as many deps as possible

### PR DESCRIPTION
**Pull request by bigdaz**

### CODE REVIEW

File: plugin/src/main/kotlin/org/gradle/github/dependency/GitHubDependencySubmissionPlugin.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        if (gradle.parent == null) {
+            LOGGER.info("Applying Plugin: GitHubDependencyExtractorPlugin to root build")
+            gradle.pluginManager.apply(GithubDependencyExtractorPlugin::class.java)
+        }
+
+        // Apply the dependency resolver to all builds
+        LOGGER.info("Applying Plugin: ForceDependencyResolutionPlugin to build $buildPath")
+        gradle.pluginManager.apply(ForceDependencyResolutionPlugin::class.java)
+    }
+}
```
Comment (JLLeitschuh) [inline plugin/src/main/kotlin/org/gradle/github/dependency/GitHubDependencySubmissionPlugin.kt:28]: EOL @ EOF

File: plugin/src/main/kotlin/org/gradle/github/dependency/extractor/ForceDependencyResolutionPlugin.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            // Depend on all 'resolveBuildDependencies' task in each included build
+            gradle.includedBuilds.forEach {
+                resolveAllDeps.dependsOn(it.task(":dependencyExtractor_resolveBuildDependencies"))
+            }
+
+            // Set the default task
+            gradle.rootProject.defaultTasks = listOf("dependencyExtractor_resolveBuildDependencies")
+        }
+    }
+}
```
Comment (JLLeitschuh) [inline plugin/src/main/kotlin/org/gradle/github/dependency/extractor/ForceDependencyResolutionPlugin.kt:26]: EOL @ EOF

File: plugin/src/main/kotlin/org/gradle/github/dependency/GitHubDependencySubmissionPlugin.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        val buildPath = (gradle as GradleInternal).identityPath
+        // Only apply the dependency extractor to the root build
+        if (gradle.parent == null) {
+            LOGGER.info("Applying Plugin: GitHubDependencyExtractorPlugin to root build")
+            gradle.pluginManager.apply(GithubDependencyExtractorPlugin::class.java)
+        }
+
+        // Apply the dependency resolver to all builds
+        LOGGER.info("Applying Plugin: ForceDependencyResolutionPlugin to build $buildPath")
+        gradle.pluginManager.apply(ForceDependencyResolutionPlugin::class.java)
```
Comment (JLLeitschuh) [inline plugin/src/main/kotlin/org/gradle/github/dependency/GitHubDependencySubmissionPlugin.kt:26]: Doesn't this prevent you from testing variants where all dependencies aren't resolved? For example, some of the existing tests that have been removed.

**Review by JLLeitschuh — CHANGES_REQUESTED** [review]

File: plugin/src/main/kotlin/org/gradle/github/dependency/GitHubDependencySubmissionPlugin.kt (hunk 3 above)
Comment (bigdaz) [reply plugin/src/main/kotlin/org/gradle/github/dependency/GitHubDependencySubmissionPlugin.kt:26]: I don't understand what you mean by "variants where all dependencies aren't resolved".

File: plugin/src/main/kotlin/org/gradle/github/dependency/extractor/ForceDependencyResolutionPlugin.kt (hunk 2 above)
Comment (bigdaz) [reply plugin/src/main/kotlin/org/gradle/github/dependency/extractor/ForceDependencyResolutionPlugin.kt:26]: This PR is not ready for review.

File: plugin/src/main/kotlin/org/gradle/github/dependency/GitHubDependencySubmissionPlugin.kt (hunk 3 above)
Comment (bigdaz) [reply plugin/src/main/kotlin/org/gradle/github/dependency/GitHubDependencySubmissionPlugin.kt:26]: The end goal for this work is #20. The `ForceDependencyResolutionPlugin` achieves this by adding a new, default task to the build that will do as much resolution as possible without requiring the project to be actually compiled and built.

I think you mean that we should continue to test that all resolved dependencies are collected when a different task is executed, one which may only resolve a subset of the build dependencies. I'll be sure to retain some coverage for this case.

**Pull request closed without merging**